test/models/user_test.rb

```diff
@@ -4,4 +4,11 @@ class UserTest < ActiveSupport::TestCase
   test "factory" do
     assert build(:user).validate!
   end
+
+  test "should not allow duplicate email addresses" do
+    create(:user, email: "test@example.com")
+    new_user = build(:user, email: "test@example.com")
+    assert_not new_user.valid?
+    assert_includes new_user.errors[:email], "has already been taken"
```

Comment:
This is a good test. It could be a good idea to add a line between the test setup and the execution.

```
email = "test@example.com"
create(:user, email:)
user = build(:user, email:)

assert_not user.valid?
assert_includes user.errors[:email], "has already been taken"
```